### PR TITLE
[libslirp] add new port

### PR DESCRIPTION
**Pull request by dhrdlicka**
**Describe the pull request**

- #### What does your PR fix?  
  Adds a [libslirp](https://gitlab.freedesktop.org/slirp/libslirp) port

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  All except non-MinGW Windows; yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by dhrdlicka** [comment]
upstream does not support MSVC, not quite sure how to mark that in the portfile or manifest

**Comment by JonLiu1993** [comment]
> upstream does not support MSVC, not quite sure how to mark that in the portfile or manifest

@dhrdlicka, If upstream does not support MSVC, please add "supports": "!windows" to vcpkg.json to indicate

**Comment by dhrdlicka** [comment]
> > upstream does not support MSVC, not quite sure how to mark that in the portfile or manifest
> 
> @dhrdlicka, If upstream does not support MSVC, please add "supports": "!windows" to vcpkg.json to indicate

The problem is that this also blocks MinGW although it is compilable on it.

**Comment by JonLiu1993** [comment]
> 

please add "supports": "!windows & !mingw" to vcpkg.json to indicate

**Comment by dg0yt** [comment]
> > 
> 
> please add "supports": "!windows & !mingw" to vcpkg.json to indicate

``!windows | mingw`

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -686,6 +686,13 @@ libressl:x64-windows        = skip
 libressl:x64-windows-static = skip
 libressl:x64-windows-static-md=skip
 libressl:x86-windows        = skip
+libslirp:arm64-windows=fail
+libslirp:arm-uwp=fail
+libslirp:x64-uwp=fail
+libslirp:x64-windows=fail
+libslirp:x64-windows-static=fail
+libslirp:x64-windows-static-md=fail
+libslirp:x86-windows=fail
```
Comment (JonLiu1993) [inline scripts/ci.baseline.txt:695]:
```suggestion
```

File: ports/libslirp/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    REF v4.6.1
+    SHA512 04a9dd88cd58c849a24b9cff405d951952760d99ea2bef0b070463dff088d79f44557a13c9427ba0043f58d4b9e06b68ff64a4f23a7b0d66df594e32e1521cae
+    HEAD_REF master
+)
+
+vcpkg_acquire_msys(MSYS_ROOT)
+vcpkg_add_to_path("${MSYS_ROOT}/usr/bin")
+
+vcpkg_configure_meson(
+    SOURCE_PATH ${SOURCE_PATH}
```
Comment (JonLiu1993) [inline ports/libslirp/portfile.cmake:14]:
```suggestion
    SOURCE_PATH “${SOURCE_PATH}”
```

File: ports/libslirp/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    OPTIONS
+        ${OPTIONS}
+)
+
+vcpkg_install_meson(ADD_BIN_TO_PATH)
+
+vcpkg_copy_pdbs()
+
+file(INSTALL ${SOURCE_PATH}/COPYRIGHT DESTINATION ${CURRENT_PACKAGES_DIR}/share/libslirp RENAME copyright)
```
Comment (JonLiu1993) [inline ports/libslirp/portfile.cmake:23]:
```suggestion
file(INSTALL “${SOURCE_PATH}/COPYRIGHT” DESTINATION “${CURRENT_PACKAGES_DIR}/share/libslirp” RENAME copyright)

```

File: ports/libslirp/vcpkg.json [hunk 4]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "libslirp",
+  "version-semver": "4.6.1",
```
Comment (JonLiu1993) [inline ports/libslirp/vcpkg.json:3]: @dhrdlicka ,Could you please confirm whether the library officially adheres to semantic versioning or not (breaking changes -> bump major version, non-breaking changes -> bump minor version, non-breaking bug fixes -> bump patch version)?

Comment (dhrdlicka) [reply ports/libslirp/vcpkg.json:3]: @JonLiu1993 yes, per the library's readme:
> We intend to use libtool's versioning for the shared libraries and use SemVer for project versions.

File: ports/libslirp/portfile.cmake (hunk 2 above)
Comment (dhrdlicka) [reply ports/libslirp/portfile.cmake]: fixed

File: ports/libslirp/portfile.cmake (hunk 3 above)
Comment (dhrdlicka) [reply ports/libslirp/portfile.cmake]: fixed

**Review by JonLiu1993 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Thanks, this LGTM!

**Pull request merged**